v2/buffer.go

```diff
@@ -0,0 +1,172 @@
+package batcher
+
+import (
+	"errors"
+	"sync"
+)
+
+type ibuffer interface {
```

Comment:
I understand why this has the "i" prefix, but it doesn't feel right. Perhaps the interface should be "buffer" and the struct "linkedBuffer". Just a suggestion though.